api/v1beta1/openstackprovisionserver.go

```diff
@@ -11,7 +12,6 @@ import (
 func GetExistingProvServerPorts(
 	ctx context.Context,
 	c goClient.Client,
-	instance *OpenStackProvisionServer,
 ) (map[string]int32, error) {
```

Comment:
If we implement this as proposed, then we can probably also simplify this data structure. We could just create a slice of port integers rather than a map.